### ISSUE TITLE
Version bumping with BREAKING CHANGES

### ISSUE THREAD

**Issue by ojizero** [lerna/lerna#1403]
When using using Lerna to publish packages with dependents in the same monorepo, Lerna tends to bump the used version in it's dependent even if the change is marked breaking (MAJOR).

## Expected Behavior

Don't bump dependents if the change in the dependency is BREAKING.

## Current Behavior

Version is bumped in the dependents when the dependency introduces a breaking changes

## Possible Solution

Providing a flag to control this behavior, such that the version isn't bumped on dependents if it affects the MAJOR of the package.

## Steps to Reproduce (for bugs)

Say for this monorepo,

    ./
    ├── functions/
    │   └── A/
    ├── lib/
    │   └── B/

and A is at version 1.0.0  and depends on B@1.0.0

and B is at version 1.0.0

When making a breaking change to B, such that B is bumped from B@1.0.0 -> to B@2.0.0

Lerna would bump used version in A as well (forces A to us B@2.0.0)
Which would conceivably break A, due to the breaking change.

<details><summary>lerna.json</summary><p>

```json
{
  "lerna": "2.8.0",
  "independent": true,
  "version": "independent",
  "packages": [
    "functions/*",
    "lib/*"
  ]
}
```
</details>

## Context

This behavior is observed in two monorepos, one is for our frontend were we share common React components, and another were we are using Lerna to manage a Microservice (AWS Lambdas) monorepo with shared libraries. In both, we are using conventional-commits for versioning.

The config provided, which is the more critical issue, is from our Microservices repo. We are using Lerna to manage linking the shared libraries (`lib` folder) and publish them to NPM, while we delegate the deployment to Apex, during the `lerna publish` step each Lambda (in the `functions` folder) could get bumped even if the change is breaking, which would trigger a release on the Lambda that may break it.

## Your Environment

| Executable | Version |
| ---: | :--- |
| `lerna --version` | 2.9.0 |
| `npm --version`  | 5.8.0 |
| `yarn --version` | N/A |
| `node --version` | v8.11.1 |

| OS | Version |
| --- | --- |
| macOS High Sierra | 10.13.4 (17E199) |

**Comment by bochen2014** [comment]
would the new versions become
```js
//A
{
   version: '1.0.1',
   dependencies:{
    B: "^2.0.0"
  }
}

// B
{
  version: "2.0.0"
}
```

**Comment by ojizero** [comment]
@bochen2014 yes exactly

**Comment by bochen2014** [comment]
not sure why this would be an issue.  if you want a `A` that depends on `B@1.0.0`, then use `A@1.0.0`;
if you don't bump `A` in your case,  you will have 2 `A@1.0.0` with one refering to B@1.0.0 and the other B@2.0.0. I don't know how are you going to differentiate them. 
You may have a genuine use case for that, but as far as I can see, that doesn't make sense to me

**Comment by evocateur** [comment]
#707 etc etc etc etc ad infinitum

**Comment by ojizero** [comment]
@evocateur @bochen2014 the issue _is not_ bumping `A`'s version, the issue is bumping the version of `B` used in `A` in the first place! This would mean that `A` is now potentially broken since it isn't guaranteed to accommodate for the changes introduced by `B`.

That is, my issue _is not_ that `A` is now `A@1.0.1`, my issue is that `A` is now using the newer version of `B` even though we can't know that this new version of `B` would retain the behaviors/APIs used by `A` (the idea being that A wasn't modified during this process).

We have no issues bumping `A` automatically (when it's dependencies are updated) when we guarantee that it won't break it, and we don't have this guarantee with a major upgrade in `B`. This is further exasperated when using CI to release new changes, especially into production environments.

**Comment by evocateur** [comment]
If these are both packages in the same monorepo, it is relatively trivial to create integration tests to ensure API compatibility. Indeed, this is one of the primary benefits of lerna in the first place.

I mean, presumably all the tests run during the development cycle that yields a breaking change in B? If they didn’t, that’s a larger problem. Publishing from CI doesn’t really change this equation, except perhaps to emphasize the importance of good tests.

**Comment by evocateur** [comment]
In that scenario, I would not recommend using a monorepo. Lerna is not a deployment tool.

**Comment by ojizero** [comment]
Indeed Lerna isn’t a deployment tool nor are we using it as such, we’re merely using it to managed shared code and resources between packages (linking them and releasing them).

And regardless of whether the dependent is a service or not, bumping it’s used version of a dependency when it receives a major upgrade would mean BREAKING that package thus regardless of it being published to NPM or deployed using other tools, this is an issue !!!

**Comment by evocateur** [comment]
> we’re merely using it to managed shared code and resources between packages

There's absolutely nothing stopping you from doing that with "normal" npm packages in different repositories. Or, for that matter, moving the services out of the monorepo and consuming the shared code as a normal dependency that one can be as sluggish about updating as you please.

> bumping it’s used version of a dependency when it receives a major upgrade would mean BREAKING that package

You mean you've never updated a major version of a dependency and it ...just worked? I mean, not all breaking changes are equal. They might have "broken" part of the API you never used, so it's by no means _necessary_ to bump your own major when you bump the majors of dependencies.

To put it another way, if you do not have sufficient tests to ensure that major releases do not break existing sibling consumers, those sibling consumers should not be located in the same monorepo.
